### PR TITLE
Add stat.sh into scripts folder

### PR DESCRIPTION
**Pull request by xChickens**
Adds a script without any conflict that allows users to check some statistics on their foundational nodes.

Basic functionality of the script:
1. Checks validation status by checking time the script is executed and compares it with the timestamp of the last bingo and sees whether or not it's within 60 seconds.
2. Prints out the balances in whichever shards the user has balance in. 
3. If run for the first time within 24 hours or if the node was reset, it saves the last bingo timestamp and last balance to the files "prevtime" and "prevbal". It also prints the difference in current balance and saved balance.
4. Calculates the estimated balance in the next 24 hours based on latest bingo timestamp and balance compared to the stored time and balance
5. Prints the timestamp of the latest bingo and the interval between the last two bingos.

### CODE REVIEW

**Comment by harmony-ek** [comment]
Jenkins speaking; can one of the admins verify this patch?

File: scripts/stat.sh [hunk 1]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+# stat.sh - Check foundational node status/statistics (requires 'bc' installed)
+
+# Get timestamps
+lastbingo=`grep -i bingo latest/*.log |tail -n 1 |cut -f 2 -d 'r' |cut -c 16-34 | tr T \ `
```
Comment (LeoHChen) [inline scripts/stat.sh]: use $() to replace all the `` to make it more uniform.

File: scripts/stat.sh [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+# stat.sh - Check foundational node status/statistics (requires 'bc' installed)
+
+# Get timestamps
+lastbingo=`grep -i bingo latest/*.log |tail -n 1 |cut -f 2 -d 'r' |cut -c 16-34 | tr T \ `
+time1=$(date -d "$lastbingo" +%s)
+rawtime=`date +%s`
+time2=$(($rawtime - 60))
+created=`grep -i -m 1 allocated latest/*.log |cut -f 7-9 -d ":" |cut -f 1 -d "." |cut -c 2- | tr T \ `
+latest=`grep -i bingo latest/*.log |tail -n 1 |cut -f 2 -d 'T' |cut -c 1-14`
+bingos=`grep -i bingo latest/*.log |tail -n 2 |cut -f 2 -d 'T' |cut -c 7-14`
```
Comment (LeoHChen) [inline scripts/stat.sh]: duplicated grep query, why don't you just use a temporary file to store all the 'grep -i bingo' ?

File: scripts/stat.sh [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+if [[ $timediff -lt 7 ]]
+  then
+    sleep $((7 - timediff))
+    balances=`./wallet.sh balances`
+  else
+    balances=`./wallet.sh balances`
+fi
+
+# Sum balances
+bal0=`echo "$balances" |grep -i "shard 0" |cut -f 2 -d ":" |cut -f1 -d "," | rev | cut -c 5- | rev`
```
Comment (LeoHChen) [inline scripts/stat.sh]: duplicated code, use an array to simplify the code.

File: scripts/stat.sh [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+reward=`echo "$curbal - $prevbal" | bc -l`
+if [[ $passed != 0 ]]
+  then
+    daily=`echo "scale=5; $reward * 86400 / $passed" | bc -l`
+  else
+    daily=0
+fi
+
+# Show results
+echo "$balances" |grep -i "address"
```
Comment (LeoHChen) [inline scripts/stat.sh]: duplicated code, use an array, and a function to simplify the duplicated code.

File: scripts/stat.sh [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+echo $daily
+
+# Find and show the time of the latest bingo
+echo -e '    Latest Bingo      :  \c' && echo $latest
+
+# Fetch the time/seconds of the last two bingos
+bingo1=`echo $bingos |cut -c 1-8`
+bingo2=`echo $bingos |cut -c 10-17`
+
+# Delete leading zero
```
Comment (LeoHChen) [inline scripts/stat.sh]: why is it needed? any example on why?

**Comment by xChickens** [comment]
Needs another review

File: scripts/stat.sh [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+bingo2=$(echo $bingos |cut -c 10-17)
+
+# Ensure bingo2 is greater than bingo 1
+if (( $(echo "$bingo2 <= $bingo1" |bc -l) ))
+    then
+        bingo2=$(echo "$bingo2 + 60" | bc -l)
+fi
+
+# Calculate and show the time difference
+difference=$(echo $bingo2-$bingo1 | bc -l)
```
Comment (LeoHChen) [inline scripts/stat.sh]: why don't you just use bash math?  Is the difference a floating number?

File: scripts/stat.sh [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+
+# Find and show the time of the latest bingo
+echo "    Latest Bingo      :  $latest"
+
+# Fetch the time/seconds of the last two bingos
+bingo1=$(echo $bingos |cut -c 1-8)
+bingo2=$(echo $bingos |cut -c 10-17)
+
+# Ensure bingo2 is greater than bingo 1
+if (( $(echo "$bingo2 <= $bingo1" |bc -l) ))
```
Comment (LeoHChen) [inline scripts/stat.sh]: why don't you compare the two number directly in bash?

File: scripts/stat.sh (hunk 6 above)
Comment (xChickens) [reply scripts/stat.sh]: Yes, bash doesn't have the ability to do floating point calculations.

File: scripts/stat.sh (hunk 7 above)
Comment (xChickens) [reply scripts/stat.sh]: Bash also can't compare two floating point numbers without use of bc.

**Comment by xChickens** [comment]
> looks good in general. maybe you could put this script into harmony-one/harmony-ops repo.

Would it be preferably to just be in the harmony-ops repo vs the main repo at all?

**Comment by LeoHChen** [comment]
Thanks for the comment.  Please put the script into harmony-ops repo then.  No other comments. @xChickens

**Pull request closed without merging**